client.go

```diff
@@ -68,6 +69,13 @@ type ClientOptions struct {
 	// multi-namenode setup (for example: 'nn/_HOST'). It is required if
 	// KerberosClient is provided.
 	KerberosServicePrincipleName string
+	// EncryptedDataNode specifies whether or not data transfer for datanodes
+	// needs to utilize encryption and thus do a negotiation to get data
+	// this is specified by dfs.encrypt.data.transfer in the config
+	EncryptedDataNode bool
+	// SecureDataNode specifies whether we're using block access tokens to
+	// communicate with datanodes, specified by dfs.block.access.token.enable in the config
+	SecureDataNode bool
```

Comment:
I thought this was something else, actually? We've always sent the block token, and I think this is just a server-side option for verifying them (with or without kerberos). In any case it should be `EnableBlockAccessToken`

Reply:
Turns out you're right, this should be changed to reference `dfs.data.transfer.protection` not `dfs.block.access.token.enable`

Reply:
Let me know if the new comments / updated name/comment is good.